adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralInterstitialRenderer.m

```diff
@@ -0,0 +1,121 @@
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    _interstitialAd = [[MTGNewInterstitialBidAdManager alloc]initWithPlacementId:placementId unitId:adUnitId delegate:self];
+    [_interstitialAd loadAdWithBidToken:adConfiguration.bidResponse];
+}
+
+- (void)newInterstitialBidAdResourceLoadSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+- (void)newInterstitialBidAdLoadFail:(nonnull NSError *)error adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+- (void)newInterstitialBidAdShowSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate willPresentFullScreenView];
+    [_adEventDelegate reportImpression];
+}
+
+- (void)newInterstitialBidAdShowFail:(nonnull NSError *)error adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate didFailToPresentWithError:error];
+}
+
+- (void)newInterstitialBidAdClicked:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate reportClick];
+}
+
+- (void)newInterstitialBidAdDismissedWithConverted:(BOOL)converted adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate willDismissFullScreenView];
+}
+
+- (void)newInterstitialBidAdDidClosed:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+#pragma mark GADMediationRewardedAd
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    
+    GADMAdapterMintegralExtras *extras = _adConfiguration.extras;
+    _interstitialAd.playVideoMute = extras.playVideoMute;
+    if ([_interstitialAd isAdReady]) {
+        [_interstitialAd showFromViewController:viewController];
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdNotValid, @"Unable to display ad.");
```

Comment:
Nit, update the error message to: `Failed to present rewarded ad from Mintegral.`

Reply:
ok,i have updated.